photutils/psf/models.py

```diff
@@ -4,17 +4,428 @@
 """
 
 from __future__ import division
+import warnings
 import numpy as np
+import copy
 from astropy.table import Table
 from astropy.modeling import models, Parameter, Fittable2DModel
 from astropy.modeling.fitting import LevMarLSQFitter
 from astropy.nddata.utils import subpixel_indices
 from ..utils import mask_to_mirrored_num
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['FittableImageModel2D', 'NonNormalizable',
+           'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model', 'get_grouped_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model',
-           'get_grouped_psf_model']
+
+class NonNormalizable(Warning):
```

Comment:
@larrybradley will know for sure, but I *think* in phoutils we've had warnings always sub-class from the astropy warning class (needed for some work-arounds of some regular warning limitations).  So that would be putting this in the imports:
```
from astropy.utils.exceptions import AstropyUserWarning
```
and having this subclass `AstropyUserWarning` instead of just `Warning`